idaes/models/flowsheets/methanol_flowsheet.py

```diff
@@ -0,0 +1,736 @@
+    m.fs.thermo_params_vapor.set_default_scaling("entr_mol", 1e-1)
+    for comp in thermo_props_vapor.config_dict["components"]:
+        m.fs.thermo_params_vapor.set_default_scaling("mole_frac_comp",
+                                                     1e2, index=comp)
+        m.fs.thermo_params_vapor.set_default_scaling("enth_mol_comp",
+                                                     1e2, index=comp)
+        m.fs.thermo_params_vapor.set_default_scaling("entr_mol_phase_comp",
+                                                     1e2, index=comp)
+        for attr in dir(getattr(m.fs.thermo_params_vapor, comp)):
+            if 'coef' in attr:
+                iscale.set_scaling_factor(getattr(
+                    getattr(m.fs.thermo_params_vapor, comp), attr), 1)
+        m.fs.thermo_params_vapor.set_default_scaling("flow_mol_phase_comp",
+                                                     1, index=comp)
+
+    m.fs.reaction_params = GenericReactionParameterBlock(
+        default={"property_package": m.fs.thermo_params_vapor,
+                 **reaction_props.config_dict})
+
+    # mixing feed streams
+    m.fs.M101 = Mixer(
+        default={"property_package": m.fs.thermo_params_vapor,
+                 "momentum_mixing_type": MomentumMixingType.minimize,
+                 "has_phase_equilibrium": True,
+                 "inlet_list": ['H2_WGS', 'CO_WGS']})
+
+    # pre-compression
+    m.fs.C101 = Compressor(
+        default={"dynamic": False,
+                 "property_package": m.fs.thermo_params_vapor,
+                 "compressor": True,
+                 "thermodynamic_assumption": ThermodynamicAssumption.isothermal
+                 })
+
+    # pre-heating
+    m.fs.H101 = Heater(
+        default={"property_package": m.fs.thermo_params_vapor,
+                 "has_pressure_change": False,
+                 "has_phase_equilibrium": False})
+
+    # reactor
+
+    m.fs.R101 = StoichiometricReactor(
+        default={"has_heat_transfer": True,
+                 "has_heat_of_reaction": True,
+                 "has_pressure_change": False,
+                 "property_package": m.fs.thermo_params_vapor,
+                 "reaction_package": m.fs.reaction_params})
+
+    # post-expansion
+    m.fs.T101 = Turbine(
+        default={"dynamic": False,
+                 "property_package": m.fs.thermo_params_vapor})
+
+    # post-cooling
+    m.fs.H102 = Heater(
+        default={"property_package": m.fs.thermo_params_vapor,
+                 "has_pressure_change": False,
+                 "has_phase_equilibrium": False})
+
+    # product recovery
+    m.fs.F101 = Flash(
+        default={"property_package": m.fs.thermo_params_VLE,
+                 "has_heat_transfer": True,
+                 "has_pressure_change": True})
+
+    # Build the flowsheet
+    print('Unit degrees of freedom')
+    for unit in ('M101', 'C101', 'H101', 'R101', 'T101', 'H102', 'F101'):
+        if unit == 'M101':
```

Comment:
This needs some comments to explain what is going on and what the expected output is. Notably, you need to explain where the `spec` value come from.